src/bin/ayb.rs

```diff
@@ -360,26 +374,58 @@ async fn main() -> std::io::Result<()> {
                 }
             }
         } else if let Some(matches) = matches.subcommand_matches("query") {
-            if let (Some(entity_database), Some(query), Some(format)) = (
+            if let (Some(entity_database), Some(format)) = (
                 matches.get_one::<EntityDatabasePath>("database"),
-                matches.get_one::<String>("query"),
                 matches.get_one::<OutputFormat>("format"),
             ) {
-                match client
+                if let Some(query) = matches.get_one::<String>("query") {
+                    match client
                     .query(&entity_database.entity, &entity_database.database, query)
                     .await
-                {
-                    Ok(query_result) => {
-                        if !query_result.rows.is_empty() {
-                            match format {
-                                OutputFormat::Table => query_result.generate_table()?,
-                                OutputFormat::Csv => query_result.generate_csv()?,
+                    {
+                        Ok(query_result) => {
+                            display_query_result(&query_result, format)?;
+                        }
+                        Err(err) => {
+                            println!("Error: {}", err);
+                        }
+                    }   
+                } else {
+                    println!("Launching an interactive session for {}/{}", entity_database.entity, entity_database.database);
+                    
+                    match DefaultEditor::new() {
```

Comment:
I'm OK with you using `?` to avoid the extra level of nesting, but I don't feel strongly :).

Reply:
Hmm, I don't know where we can exactly use a `?` there.

Reply:
I suspect we can drop the `match` and call `DefaultEditor::new()?`, but will try it after merging :)